examples/cpp_sycl/source/decision_forest/df_reg_dense_batch.cpp

```diff
@@ -0,0 +1,144 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+!  Content:
+!    C++ example of decision forest regression in the batch processing mode.
+!
+!    The program trains the decision forest regression model on a training
+!    datasetFileName and computes regression for the test data.
+!******************************************************************************/
+
+/**
+ * <a name="DAAL-EXAMPLE-CPP-DF_REG_DENSE_BATCH"></a>
+ * \example df_reg_dense_batch.cpp
+ */
+
+#include "daal_sycl.h"
+#include "service.h"
+#include "service_sycl.h"
+
+using namespace std;
+using namespace daal;
+using namespace daal::data_management;
+using namespace daal::algorithms::decision_forest::regression;
+
+/* Input data set parameters */
+const string trainDatasetFileName         = "../data/batch/df_regression_train.csv";
+const string testDatasetFileName          = "../data/batch/df_regression_test.csv";
+const size_t categoricalFeaturesIndices[] = { 3 };
+const size_t nFeatures                    = 13; /* Number of features in training and testing data sets */
+
+/* Decision forest parameters */
+const size_t nTrees = 100;
+
+training::ResultPtr trainModel();
+void testModel(const training::ResultPtr & res);
+void loadData(const std::string & fileName, NumericTablePtr & pData, NumericTablePtr & pDependentVar);
+
+int main(int argc, char * argv[])
+{
+    checkArguments(argc, argv, 2, &trainDatasetFileName, &testDatasetFileName);
+
+    for (const auto & deviceSelector : getListOfDevices())
+    {
+        const auto & nameDevice = deviceSelector.first;
+        const auto & device     = deviceSelector.second;
+        cl::sycl::queue queue(device);
+        std::cout << "Running on " << nameDevice << "\n\n";
+
+        daal::services::SyclExecutionContext ctx(queue);
+        services::Environment::getInstance()->setDefaultExecutionContext(ctx);
+
+        training::ResultPtr trainingResult = trainModel();
+        testModel(trainingResult);
+    }
+    return 0;
+}
+
+training::ResultPtr trainModel()
+{
+    /* Create Numeric Tables for training data and dependent variables */
+    NumericTablePtr trainData;
+    NumericTablePtr trainDependentVariable;
+
+    loadData(trainDatasetFileName, trainData, trainDependentVariable);
+
+    /* Create an algorithm object to train the decision forest regression model with the default method */
+    training::Batch<> algorithm;
+
+    /* Pass a training data set and dependent values to the algorithm */
+    algorithm.input.set(training::data, trainData);
+    algorithm.input.set(training::dependentVariable, trainDependentVariable);
+
+    algorithm.parameter.nTrees           = nTrees;
+    algorithm.parameter.varImportance    = daal::algorithms::decision_forest::training::MDA_Raw;
```

Comment:
Do you support all of these parameters? I would not recommend adding to the examples what we do not currently support.

Reply:
yes, all of them are supported